frontend/public/components/hpa.tsx

```diff
@@ -31,8 +34,113 @@ const menuActions = [
   ...common,
 ];
 
-export const HorizontalPodAutoscalersDetails: React.SFC<HorizontalPodAutoscalersDetailsProps> = ({obj: hpa}) => {
-  return <div className="co-m-pane__body">
+const MetricsRow: React.SFC<MetricsRowProps> = ({type, current, target}) => <div className="row">
+  <div className="col-xs-6">
+    {type}
+  </div>
+  <div className="col-xs-3">
+    {current || '-'}
+  </div>
+  <div className="col-xs-3">
+    {target || '-'}
+  </div>
+</div>;
+
+const externalRow = (metric, current, key) => {
+  const { external } = metric;
+  const type = external.metricName;
+
```

Comment:
Thoughts on splitting these apart?

```JavaScript 
// vs 
const currentValue = external.targetAverageValue
  ? _.get(current, 'object.currentAverageValue') 
  : _.get(current, 'object.currentValue');

const targetValue = external.targetAverageValue
  ? external.targetAverageValue
  : external.targetValue;
```
is a nit, but a little more declarative (I'm using the `ternary` the way I see it used elsewhere in the codebase).